src/commands/pullComment.ts

```diff
@@ -0,0 +1,73 @@
+import { PermissionFlagsBits, SlashCommandBuilder } from "discord.js";
+
+import { PullComments } from "../config/PullComments";
+import { PrivilegedCommand } from "../interfaces/PrivilegedCommand";
+import { errorHandler } from "../utils/errorHandler";
+
+export const pullComment: PrivilegedCommand = {
+  requiredPermissions: [PermissionFlagsBits.ModerateMembers],
+  guildOnly: true,
+  data: new SlashCommandBuilder()
+    .setName("pull-comment")
+    .setDescription("Adds a friendly comment to a pull request.")
+    .addStringOption((option) =>
+      option
+        .setName("repository")
+        .setDescription(
+          "The URL-friendly name of the repository to comment on."
+        )
+        .setRequired(true)
+    )
+    .addIntegerOption((option) =>
+      option
+        .setName("number")
+        .setDescription("The number of the pull request to comment on.")
+        .setRequired(true)
+    )
+    .addStringOption((option) =>
+      option
+        .setName("message")
+        .setDescription("The message to post.")
+        .setRequired(true)
+        .addChoices(...PullComments.map((c) => ({ name: c.key, value: c.key })))
+    ),
```

Comment:
Any chance we can mimic the `gh` cli api:
```bash
gh pr [-R [HOST/]OWNER/REPO] comment [<number> | <url> | <branch>] [-b text]
```

I am not sure what that would look like in terms of Discourse. Come to look at it again, I am not sure what this currently looks like.

Reply:
This is Discord, not discourse. So it's a slash command, not a text command.

Reply:
🤦‍♂️ It has been a long day. As you were.

Reply:
<img width="324" alt="image" src="https://github.com/freeCodeCamp/discord-bot/assets/63889819/642a55a2-784a-43a3-bc99-07327357e696">

Reply:
Does Discord have anything like sub-commands? Or, would that all have to be string values then matched?

Reply:
We can have subcommands, but what would be the intended UX here?

Reply:
Well, it is definitely not necessary if all we ever want is this one instruction. If we wanted more things like:

```bash
GENERAL COMMANDS
  create:      Create a pull request
  list:        List pull requests in a repository
  status:      Show status of relevant pull requests

TARGETED COMMANDS
  checkout:    Check out a pull request in git
  checks:      Show CI status for a single pull request
  close:       Close a pull request
  comment:     Add a comment to a pull request
  diff:        View changes in a pull request
  edit:        Edit a pull request
  lock:        Lock pull request conversation
  merge:       Merge a pull request
  ready:       Mark a pull request as ready for review
  reopen:      Reopen a pull request
  review:      Add a review to a pull request
  unlock:      Unlock pull request conversation
  view:        View a pull request
```

Then, it would be nicer.

Reply:
I'm not sure how a Discord bot would check out a PR haha.

No, the purpose of this is solely so we can comment on PRs through Camperbot's account (to keep things anonymous and less awkward).